CostumersAPI/Controllers/CostumersController.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using Microsoft.AspNetCore.Mvc;
+using CostumersAPI.Costumer;
+using CostumersAPI.Services.Interfaces;
+using FluentValidation;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Http;
+
+namespace CostumersAPI.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CostumersController : ControllerBase
+    {
+        private readonly ICostumerService _costumerService;
+        private readonly ILogger<CostumersController> _logger;
+
+        public CostumersController(ICostumerService costumerService, ILogger<CostumersController> logger)
+        {
+            if (costumerService != null)
+                _costumerService = costumerService;
```

Comment:
A ideia da validação de nulo é nos informar que algum serviço não foi possível fazer injeção de dependência e não mascarar o erro para apenas injetar quando não for nulo, pois dessa forma irá estourar um erro pior e mais difícil de entender
```suggestion
                _costumerService = costumerService ?? throw new ArgumentNullException(nameof(costumerService));
```